### ISSUE TITLE
error on btrbk list latest

### ISSUE THREAD

**Issue by ian-kelling** [digint/btrbk#111]
```
$ btrbk list latest

ERROR: process died unexpectedly (btrbk v0.24.0-dev)
Please contact the author: Axel Burri <axel@tty0.ch>

Stack Trace:
----------------------------------------
Use of uninitialized value in numeric eq (==) at ./btrbk line 4773.
 at ./btrbk line 253.
	main::__ANON__("Use of uninitialized value in numeric eq (==) at ./btrbk line"...) called at ./btrbk line 4773

```

**Comment by ian-kelling** [comment]
debug output, same error:

```
./btrbk -l debug list latest
btrbk command line client, version 0.24.0-dev  (Wed Nov  9 13:01:44 2016)
Using configuration: /etc/btrbk.conf
Initializing volume section: /mnt/iroot
### btrfs subvolume show /mnt/iroot
Command execution successful
found btrfs root: /mnt/iroot
### btrfs subvolume list -a -c -u -q -R /mnt/iroot
Command execution successful
Parsed 7 total subvolumes for filesystem at: /mnt/iroot
### btrfs subvolume list -a -r /mnt/iroot
Command execution successful
Parsed 6 readonly subvolumes for filesystem at: /mnt/iroot
Initializing subvolume section: /mnt/iroot/i
Found "/mnt/iroot/i" (id=3430) in btrfs subtree of: /mnt/iroot
Initializing volume section: /mnt/root
### btrfs subvolume show /mnt/root
Command execution successful
found btrfs root: /mnt/root
### btrfs subvolume list -a -c -u -q -R /mnt/root
Command execution successful
Parsed 24 total subvolumes for filesystem at: /mnt/root
### btrfs subvolume list -a -r /mnt/root
Command execution successful
Parsed 17 readonly subvolumes for filesystem at: /mnt/root
Initializing subvolume section: /mnt/root/q
Found "/mnt/root/q" (id=833) in btrfs subtree of: /mnt/root
Initializing volume section: /mnt/root
Initializing subvolume section: /mnt/root/p
Found "/mnt/root/p" (id=1595) in btrfs subtree of: /mnt/root
Initializing target section: frodo:/mnt/root/treetowl-btrbk
### ssh -i /root/.ssh/id_rsa root@frodo 'btrfs subvolume show /mnt/root/treetowl-btrbk'
Command execution failed (exitcode=1): "ssh -i /root/.ssh/id_rsa root@frodo 'btrfs subvolume show /mnt/root/treetowl-btrbk' 2>&1"
Resolving btrfs mount point for: frodo:/mnt/root/treetowl-btrbk
### ssh -i /root/.ssh/id_rsa root@frodo 'readlink -e /mnt/root/treetowl-btrbk'
Command execution successful
Real path for "frodo:/mnt/root/treetowl-btrbk" is: /mnt/root/treetowl-btrbk
### ssh -i /root/.ssh/id_rsa root@frodo 'cat /proc/self/mounts'
Command execution successful
Found btrfs mount point for "frodo:/mnt/root/treetowl-btrbk": /mnt/root (subvolid=5)
### ssh -i /root/.ssh/id_rsa root@frodo 'btrfs subvolume list -a -c -u -q -R /mnt/root'
Command execution successful
Parsed 11 total subvolumes for filesystem at: frodo:/mnt/root
### ssh -i /root/.ssh/id_rsa root@frodo 'btrfs subvolume list -a -r /mnt/root'
Command execution successful
Parsed 9 readonly subvolumes for filesystem at: frodo:/mnt/root
Initializing target section: frodo:/mnt/root/treetowl-btrbk
### ssh -i /root/.ssh/id_rsa root@frodo 'btrfs subvolume show /mnt/root/treetowl-btrbk'
Command execution failed (exitcode=1): "ssh -i /root/.ssh/id_rsa root@frodo 'btrfs subvolume show /mnt/root/treetowl-btrbk' 2>&1"
Resolving btrfs mount point for: frodo:/mnt/root/treetowl-btrbk
Found btrfs mount point for "frodo:/mnt/root/treetowl-btrbk": /mnt/root (subvolid=5)
Initializing target section: frodo:/mnt/root/treetowl-btrbk
### ssh -i /root/.ssh/id_rsa root@frodo 'btrfs subvolume show /mnt/root/treetowl-btrbk'
Command execution failed (exitcode=1): "ssh -i /root/.ssh/id_rsa root@frodo 'btrfs subvolume show /mnt/root/treetowl-btrbk' 2>&1"
Resolving btrfs mount point for: frodo:/mnt/root/treetowl-btrbk
Found btrfs mount point for "frodo:/mnt/root/treetowl-btrbk": /mnt/root (subvolid=5)
Found 5 snapshot children of "/mnt/iroot/i" in: /mnt/iroot
Latest common subvolumes for: /mnt/iroot/i: src=/mnt/iroot/i.20161101T201346-0700  target=frodo:/mnt/root/treetowl-btrbk/i.20161101T201346-0700
Found 14 snapshot children of "/mnt/root/q" in: /mnt/root
Latest common subvolumes for: /mnt/root/q: src=/mnt/root/q.20161101T201346-0700  target=frodo:/mnt/root/treetowl-btrbk/q.20161101T201346-0700
Found 0 snapshot children of "/mnt/root/p" in: /mnt/root
No common subvolumes of "/mnt/root/p" found in src="/mnt/root/", target="frodo:/mnt/root/treetowl-btrbk/"
Found 0 snapshot children of "/mnt/root/p" in: /mnt/root
No latest snapshots found for: /mnt/root/p

ERROR: process died unexpectedly (btrbk v0.24.0-dev)
Please contact the author: Axel Burri <axel@tty0.ch>

Stack Trace:
----------------------------------------
Use of uninitialized value in numeric eq (==) at ./btrbk line 4773.
 at ./btrbk line 253.
	main::__ANON__("Use of uninitialized value in numeric eq (==) at ./btrbk line"...) called at ./btrbk line 4773
```

**Comment by digint** [comment]
Thanks for reporting!

Bug triggers when there are no snapshots at all.

fixed in: 57be688f6b83189828e2959f14f1ad6cedd0fa73
